packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2062,24 +2082,43 @@ export class TransactionController extends BaseController<
   /**
    * Update a custodial transaction.
    *
-   * @param transactionId - The ID of the transaction to update.
-   * @param options - The custodial transaction options to update.
-   * @param options.errorMessage - The error message to be assigned in case transaction status update to failed.
-   * @param options.hash - The new hash value to be assigned.
-   * @param options.status - The new status value to be assigned.
+   * @param payload - The custodial transaction options to update.
+   * @param payload.transactionId - The ID of the transaction to update.
+   * @param payload.errorMessage - The error message to be assigned in case transaction status update to failed.
+   * @param payload.hash - The new hash value to be assigned.
+   * @param payload.status - The new status value to be assigned.
+   * @param payload.gasLimit - The new gas limit value to be assigned
+   * @param payload.gasPrice - The new gas price value to be assigned
+   * @param payload.maxFeePerGas - The new max fee per gas value to be assigned
+   * @param payload.maxPriorityFeePerGas - The new max priority fee per gas value to be assigned
+   * @param payload.nonce - The new nonce value to be assigned
+   * @param payload.type - The tranasction type (hardfork) to be assigned
    */
-  updateCustodialTransaction(
-    transactionId: string,
-    {
-      errorMessage,
-      hash,
-      status,
-    }: {
-      errorMessage?: string;
-      hash?: string;
-      status?: TransactionStatus;
-    },
-  ) {
+  updateCustodialTransaction({
+    transactionId,
```

Comment:
As there is lots of properties here, could we create a type in `types.ts` such as `UpdateCustodialTransactionRequest` to make it more readable and migrate the JSDoc to the type itself?

Reply:
Excellent point, done